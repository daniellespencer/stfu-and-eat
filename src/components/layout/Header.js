import React, { Fragment, useContext } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Navbar, Nav } from "react-bootstrap";
import { AuthContext } from "../auth/context/authContext";

const Header = ({ title, icon }) => {
  const authContext = useContext(AuthContext);

  const { isAuthenticated, logout, user } = authContext;

  const onLogout = () => {
    logout();
  };

  const authLinks = (
    <Fragment>
      <li>Hello {user && user.name}</li>
      <Nav.Link as={Link} to='recommendation'>
        Recommendation
      </Nav.Link>
      <Nav.Link as={Link} to='restaurant-list'>
        Restaurant List
      </Nav.Link>
      <Nav.Link as={Link} to='about'>
        About
      </Nav.Link>
      <Nav.Link as={Link} onClick={onLogout} to='landing'>
        <i className='fa fa-sign-out'></i>{" "}
        <span className='hide-sm'>Logout</span>
      </Nav.Link>
    </Fragment>
  );

  const guestLinks = (
    <Fragment>
      <Nav.Link as={Link} to='recommendation'>
        Recommendation
      </Nav.Link>
      <Nav.Link as={Link} to='login'>
        Login
        {' '}<i className='fa fa-sign-in'></i>
      </Nav.Link>
    </Fragment>
  );

  return (
    <Navbar bg='dark' variant='dark' expand='lg'>
      <Navbar.Brand as={Link} to='/'>
        <i className={icon}></i> {title}
      </Navbar.Brand>
      <Navbar.Toggle aria-controls='basic-navbar-nav' />
      <Navbar.Collapse id='basic-navbar-nav'>
        <Nav className='ml-auto'>
          {/* add condition here? */}
          {isAuthenticated ? authLinks : guestLinks}
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

Header.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string,
};

Header.defaultProps = {
  title: "STFU & EAT",
  icon: "fa fa-utensils",
};

export default Header;
