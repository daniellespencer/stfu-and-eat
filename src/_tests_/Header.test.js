import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';

import Header from '../components/layout/Header';

afterEach(cleanup);

it("renders correctly", () => {
    shallow(<Header />);
})