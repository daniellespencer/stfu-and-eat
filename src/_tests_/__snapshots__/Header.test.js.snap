// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders Header with user links when authenticated 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <i
      className="far fa-comment-alt"
    />
     
    STFU & EAT
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/recommendation"
        onClick={[Function]}
      >
        Recommendation
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/restaurant-list"
        onClick={[Function]}
      >
        Restaurant List
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/landing"
        onClick={[Function]}
      >
        <i
          className="fa fa-sign-out"
        />
         
        <span
          className="hide-sm"
        >
          Logout
        </span>
      </a>
    </div>
  </div>
</nav>
`;

exports[`Header Renders Header with user links when unauthenticated 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <i
      className="far fa-comment-alt"
    />
     
    STFU & EAT
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/recommendation"
        onClick={[Function]}
      >
        Recommendation
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/login"
        onClick={[Function]}
      >
        Login
         
        <i
          className="fa fa-sign-in"
        />
      </a>
    </div>
  </div>
</nav>
`;
